lxd/images.go

```diff
@@ -1115,93 +1115,214 @@ func autoUpdateImagesTask(d *Daemon) (task.Func, task.Schedule) {
 }
 
 func autoUpdateImages(ctx context.Context, d *Daemon) error {
-	projectNames := []string{}
+	imageMap := make(map[string][]db.Image)
+
 	err := d.cluster.Transaction(func(tx *db.ClusterTx) error {
-		projects, err := tx.GetProjects(db.ProjectFilter{})
+		var err error
+
+		images, err := tx.GetImages(db.ImageFilter{AutoUpdate: true})
 		if err != nil {
 			return err
 		}
 
-		for _, project := range projects {
-			projectNames = append(projectNames, project.Name)
+		for _, image := range images {
+			imageMap[image.Fingerprint] = append(imageMap[image.Fingerprint], image)
 		}
 
 		return nil
 	})
 	if err != nil {
-		return errors.Wrap(err, "Unable to retrieve project names")
+		return errors.Wrap(err, "Unable to retrieve image fingerprints")
 	}
 
-	for _, project := range projectNames {
-		err := autoUpdateImagesInProject(ctx, d, project)
+	for fingerprint, images := range imageMap {
+		nodes, err := d.cluster.GetNodesWithImageAndAutoUpdate(fingerprint, true)
 		if err != nil {
-			return errors.Wrapf(err, "Unable to update images for project %s", project)
+			logger.Debug("Error getting nodes", log.Ctx{"fingerprint": fingerprint})
```

Comment:
@monstermunchkin same point as below about use of "nodes" rather than "cluster members" in log message.

Reply:
Also here, if you're changing it, might be worth expanding the log message to say something like "Error getting cluster members for image auto update" to give the log message a bit more context (as otherwise apart from inferring it from fingerprint also logged, we wouldn't know the log message is related to images, or the specific operation on them).

Reply:
This should be a logger.Error right?